src/algorithms/trie.cpp

```diff
@@ -0,0 +1,77 @@
+#include "trie.hpp"
+#include <iostream>
+#include <cassert>
+
+Trie::Node::Node() {
+	children.fill(nil);
+}
+
+Trie::Trie (){
+	nodes.push_back(Node {});
+}
+
+uint16_t Trie::go(uint16_t n, uint8_t uc) {
+	return nodes[n].children[uc];
+}
+
+void Trie::insert(Entry entry) {
+	int entry_idx = entries.size();
+	entries.push_back(entry);
+
+	int n = 0;
+	for (char c : entry.text) {
+		uint8_t uc = c;
+		auto nn = go(n, uc);
```

Comment:
Perhaps more purpose indicative identifiers would be nice here.

Reply:
And everywhere for that matter.

Reply:
Lol it's very competitive programming-y

Reply:
FWIW, c stands for char, uc for unsigned char, n for node, and nn for next node.

I'll change the names